src/neuroglancer/datasource/computed/frontend.ts

```diff
@@ -0,0 +1,258 @@
+      chunkManager: ChunkManager,
+      cancellationToken: CancellationToken): Promise<ComputedVolumeSpecs> {
+    return new Promise((resolve, reject) => {
+             const dataSource = dataSourceProvider.getDataSource(originUrl)[0];
+             if (!dataSource || !dataSource.getVolume) {
+               reject();
+             }
+             resolve(dataSource.getVolume!(chunkManager, config, {}, cancellationToken));
+           })
+        .then((multiScaleVolumeChunkSource: GenericMultiscaleVolumeChunkSource) => {
+          const sources = multiScaleVolumeChunkSource.getSources({});
+          const specs = sources.map((volumeChunkSources) => {
+            return volumeChunkSources.map((volumeChunkSource) => {
+              return volumeChunkSource.spec;
+            });
+          });
+
+          return {
+            specs,
+            sources,
+            volumeType: multiScaleVolumeChunkSource.volumeType,
+            dataType: multiScaleVolumeChunkSource.dataType,
+            numChannels: multiScaleVolumeChunkSource.numChannels,
+          };
+        });
+  }
+
+  /**
+   * Creates a ComputedVolumeDataSourceParameters object with default values
+   * populated from the first volume spec at native resolution.
+   * @param volumeSpecs ComputedVolumeSpecs
+   * @param originUrl url for the origin data source
+   * @param dataSourceProvider
+   * @returns ComputedVolumeDataSourceParameters
+   */
+  defaultParams(
+      volumeSpecs: ComputedVolumeSpecs, originUrl: string,
+      dataSourceProvider: DataSourceProvider): ComputedVolumeDataSourceParameters {
+    const spec = volumeSpecs.sources[0][0].spec;
+    // Default DataType, VolumeType, channel count
+    const {dataType, volumeType, numChannels} = volumeSpecs;
+    // Default chunk size, used for input and output computation buffer sizes.
+    const size = spec.chunkDataSize;
+    return {
+      originUrl,
+      computationParameters: {
+        inputSpec: {size: vec3.copy(vec3.create(), size), dataType, volumeType, numChannels},
+        outputSpec: {size: vec3.copy(vec3.create(), size), dataType, volumeType, numChannels}
+      },
+      inputScaleIndex: 0,
+      inputSourceIndex: 0,
+      outputVoxelSize: vec3.copy(vec3.create(), spec.voxelSize),
+      outputVolumeSize: vec3.copy(vec3.create(), spec.upperVoxelBound),
+      outputVoxelOffset: vec3.copy(vec3.create(), spec.baseVoxelOffset),
+      dataSourceProvider
+    };
+  }
+
+  get description() {
+    return 'Computed data source.';
+  }
+
+  getVolume(
+      chunkManager: ChunkManager, config: string, options: GetVolumeOptions,
+      cancellationToken: CancellationToken) {
+    if (!options.dataSourceProvider) {
+      return Promise.reject('Need a DataSourceProvider');
+    }
+
+    const dataSourceProvider = options.dataSourceProvider!;
+    const [sourceNames, sourceConfigs] = this.parseConfig(config, dataSourceProvider);
+
+    if (sourceNames.length < 2) {
+      return Promise.reject();
```

Comment:
It would be good to reject with an informative error message using: reject(new Error("Message here")).

Reply:
(That applies to elsewhere as well.)

Reply:
Done.